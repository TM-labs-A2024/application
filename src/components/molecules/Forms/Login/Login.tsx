import { Button } from '@chakra-ui/react'
import Input from '@components/atoms/Input'
import React from 'react'

export default function LoginForm({ isLoading }: { isLoading: boolean }) {
  return (
    <form className="flex flex-col items-center gap-4" data-testid="login-form">
      <Input
        placeholder="Introduzca su correo electr칩nico"
        label="Correo electr칩nico"
        className="w-full"
        disabled={isLoading}
        id="email"
      />
      <Input
        placeholder="Introduzca su contrase침a"
        label="Contrase침a"
        className="w-full"
        disabled={isLoading}
        id="password"
      />
      <Button isLoading={isLoading} type="submit">
        Ingresar
      </Button>
    </form>
  )
}
