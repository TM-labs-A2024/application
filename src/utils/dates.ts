export const formatDate = (date: string) => {
  const chunks = date.split('-')

  const newDate = chunks[1] + '-' + chunks[2] + '-' + chunks[0]

  return newDate
}
