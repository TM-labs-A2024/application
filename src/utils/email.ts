import emailjs from '@emailjs/browser'

export function sendEmail(data: {
  from_name: string
  to_name: string
  code: string | number
  to_email: string
}) {
  const templateParams = {
    ...data
  }

  const serviceKey = process.env.NEXT_PUBLIC_EMAIL_SERVICE_KEY ?? ''
  const templateKey = process.env.NEXT_PUBLIC_EMAIL_TEMPLATE_KEY ?? ''
  const apiKey = process.env.NEXT_PUBLIC_EMAIL_API_KEY ?? ''

  emailjs.send(serviceKey, templateKey, templateParams, apiKey).catch(() => {
    alert('Algo sali√≥ mal, por favor reintenta el registro.')
  })
}
