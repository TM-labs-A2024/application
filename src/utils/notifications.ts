import { GENERIC_ERROR } from '@constants/index'
import { isMobile } from '@utils/index'
import { Store } from 'react-notifications-component'

export const setupErrorNotification = () => {
  Store.addNotification(
    GENERIC_ERROR('Algo salió mal, por favor inténtelo de nuevo.', isMobile(window))
  )
}
