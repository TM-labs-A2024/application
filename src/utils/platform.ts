import { Capacitor } from '@capacitor/core'

export const isWeb = () => Capacitor.getPlatform() === 'web'

export const isIOS = () => Capacitor.getPlatform() === 'ios'

export const isAndroid = () => Capacitor.getPlatform() === 'android'
