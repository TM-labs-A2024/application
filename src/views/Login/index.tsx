import { Link } from '@chakra-ui/react'
import LoginForm from '@components/molecules/Forms/Login'
import Splash from '@src/components/atoms/Splash'
import { Login } from '@src/types'
import NextLink from 'next/link'
import React from 'react'

import Logo from '../../../public/static/icons/logo.svg'

export default function LoginView({
  context
}: {
  context: {
    isLoading: boolean
    type: string
    loginPatient: (arg: Login) => void
    loginDoctor: (arg: Login) => void
    loginNurse: (arg: Login) => void
    loginInstitution: (arg: Login) => void
    loginGovernment: (arg: Login) => void
  }
}) {
  return context.isLoading ? (
    <Splash />
  ) : (
    <div className="flex h-screen w-screen flex-col items-center justify-between p-24">
      <Logo />
      <LoginForm context={context} />
      <Link as={NextLink} href="/registro">
        Crear cuenta
      </Link>
    </div>
  )
}
