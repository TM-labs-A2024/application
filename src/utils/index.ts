import { formatDate } from './dates'
import { sendEmail, generateVerificationCode, setupEmailSending } from './email'
import { setupErrorNotification } from './notifications'
import { isWeb, isIOS, isAndroid, isMobile } from './platform'

export {
  isWeb,
  isIOS,
  isAndroid,
  isMobile,
  sendEmail,
  generateVerificationCode,
  setupEmailSending,
  setupErrorNotification,
  formatDate
}
