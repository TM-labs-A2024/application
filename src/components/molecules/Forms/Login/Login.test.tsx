import { render, screen } from '@test/utils'

import Login from './Login'
import '@testing-library/jest-dom'

describe('Molecules > Forms > Login test', () => {
  test('The component requires the fields: Email and password', () => {
    render(<Login isLoading={false} />)

    expect(screen.getByLabelText('Correo electr칩nico')).toBeInTheDocument()
    expect(screen.getByLabelText('Contrase침a')).toBeInTheDocument()
  })

  test('The component shows the respective placeholder', () => {
    render(<Login isLoading={false} />)

    expect(screen.getByPlaceholderText('Introduzca su correo electr칩nico')).toBeInTheDocument()
    expect(screen.getByPlaceholderText('Introduzca su contrase침a')).toBeInTheDocument()
  })

  test('Matches the snapshot', () => {
    render(<Login isLoading={false} />)

    expect(screen.getByTestId('login-form')).toMatchSnapshot()
  })
})
