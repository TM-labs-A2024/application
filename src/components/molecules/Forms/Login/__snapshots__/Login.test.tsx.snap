// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules > Forms > Login test Matches the snapshot 1`] = `
<form
  class="flex flex-col items-center gap-4"
  data-testid="login-form"
>
  <div
    class="w-full"
    data-testid="input-component"
  >
    <label
      for="email"
    >
      Correo electr칩nico
    </label>
    <input
      class="chakra-input min-h-10 css-1cjy4zv"
      id="email"
      placeholder="Introduzca su correo electr칩nico"
    />
  </div>
  <div
    class="w-full"
    data-testid="input-component"
  >
    <label
      for="password"
    >
      Contrase침a
    </label>
    <input
      class="chakra-input min-h-10 css-1cjy4zv"
      id="password"
      placeholder="Introduzca su contrase침a"
    />
  </div>
  <button
    class="chakra-button css-ez23ye"
    type="submit"
  >
    Ingresar
  </button>
</form>
`
